joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/PythonTypeRecovery.scala

```diff
@@ -0,0 +1,370 @@
+package io.joern.x2cpg.passes.frontend
+
+import io.joern.x2cpg.Defines
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.{Operators, PropertyNames}
+import io.shiftleft.passes.CpgPass
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes.Assignment
+import org.apache.logging.log4j.LogManager
+
+import java.io.{File => JFile}
+import java.util.concurrent.RecursiveTask
+import java.util.regex.Matcher
+import scala.collection.MapView
+import scala.collection.concurrent.TrieMap
+import scala.util.Try
+
+/** Based on a flow-insensitive symbol-table-style approach. This does not accurately determine the difference between
+  * shadowed variables in the same file but this is due to REF edges not connecting children methods to parent scope
+  * (yet).
+  *
+  * TODO: Currently this just improves the type recovery of imported function and types.
+  */
+class PythonTypeRecovery(cpg: Cpg) extends CpgPass(cpg) {
```

Comment:
Same comment as for the above pass: can we do this in parallel for multiple independent units?